### PR TITLE
feat: add components [5]

### PR DESCRIPTION
**Pull request by alexruzenhack**
### Motivation

These components are introduced to be used in the context of Nano Contract UI implementations and refactorings.

### Acceptance Criteria
- feat: add HathorFlatList component
- feat: add TransactionStatusLabel component

>[!NOTE]
>It depends on:
>- https://github.com/HathorNetwork/hathor-wallet-mobile/pull/453

### Security Checklist
- [x] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: src/components/ActionDot.icon.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/**
```
Comment (andreabadesso) [inline src/components/ActionDot.icon.js:1]: This is also being added in the `add icons` [PR](https://github.com/HathorNetwork/hathor-wallet-mobile/pull/453)

Should you change the base branch?

File: src/components/HathorFlatList.component.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    <FlatList
+      ItemSeparatorComponent={ItemSeparator}
+      // Introduced last to allow overwrite
+      {...props}
+    />
+  </View>
+);
+
+const ItemSeparator = () => (
+  <View style={{ width: '100%', height: 1, backgroundColor: COLORS.borderColor }} />
```
Comment (andreabadesso) [inline src/components/HathorFlatList.component.js:28]: Move those styles to the stylesheet you created below

File: src/components/HathorFlatList.component.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import {
+  FlatList, FlatListProps, StyleSheet, View,
```
Comment (andreabadesso) [inline src/components/HathorFlatList.component.js:10]: `FlatListProps` not being used as per the warning

**Review by andreabadesso — CHANGES_REQUESTED** [review]

File: src/components/HathorFlatList.component.js (hunk 2 above)
Comment (alexruzenhack) [reply src/components/HathorFlatList.component.js:28]: It was moved.

File: src/components/HathorFlatList.component.js (hunk 3 above)
Comment (alexruzenhack) [reply src/components/HathorFlatList.component.js:10]: It was removed.

File: src/components/TransactionStatusLabel.component.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    <CircleError size={16} color={styles.freeze.color} />
+  </TransactionStatusBase>
+);
+
+/**
+ * @param {Object} param
+ * @param {'confirmed'|'processing'} param.processingStatus Status of a Nano Contract transaction
+ * @param {boolean} param.isVoided Transaction's void flag
+ */
+export const TransactionStatusLabel = ({ processingStatus, isVoided = false }) => {
```
[the remark this answers is not in the document]
Comment (andreabadesso) [reply src/components/TransactionStatusLabel.component.js:96]: Why?

[the remark this answers is not in the document]
Comment (alexruzenhack) [reply src/components/TransactionStatusLabel.component.js]: I refactored it.

**Review by andreabadesso — APPROVED** [review]

File: src/sagas/networkSettings.js [hunk 5]
```diff
@@ -176,7 +171,7 @@ export function* updateNetworkSettings(action) {
         potentialNetwork = response;
       }
     } catch (err) {
-      log.error('Error calling the wallet-service while trying to get network details in updateNetworkSettings effect.', err);
+      console.error('Error calling the wallet-service while trying to get network details in updateNetworkSettings effect.', err);
```
Comment (andreabadesso) [inline src/sagas/networkSettings.js:174]: Why did you stop using log here?

File: src/sagas/tokens.js [hunk 6]
```diff
@@ -249,7 +239,7 @@ export function* fetchTokenMetadata({ tokenId }) {
       tokenId,
     });
     // eslint-disable-next-line
-    log.log('Error downloading metadata of token', tokenId);
+    console.log('Error downloading metadata of token', tokenId);
```
Comment (andreabadesso) [inline src/sagas/tokens.js:242]: Why?

**Review by andreabadesso — CHANGES_REQUESTED** [review]

File: src/sagas/networkSettings.js (hunk 5 above)
Comment (alexruzenhack) [reply src/sagas/networkSettings.js]: Resolved

File: src/sagas/tokens.js (hunk 6 above)
Comment (alexruzenhack) [reply src/sagas/tokens.js:242]: Resolved

**Review by andreabadesso — APPROVED** [review]

File: locale/pt-br/texts.po [hunk 7]
```diff
@@ -1160,6 +1160,18 @@ msgid ""
...
+msgid "Executed"
+msgstr "Executada"
+
+#: src/components/TransactionStatusLabel.js:76
+msgid "Processing"
+msgstr "Processando"
+
+#: src/components/TransactionStatusLabel.js:81
+msgid "Voided"
+msgstr "Invalidada"
```
Comment (pedroferreira1) [inline locale/pt-br/texts.po:1173]: Why not "Inválida"?

File: src/sagas/wallet.js [hunk 8]
```diff
@@ -320,7 +320,7 @@ export function* loadTokens() {
   // spawn a new "thread" to handle it.
   //
   // `spawn` is similar to `fork`, but it creates a `detached` fork
-  yield spawn(fetchTokensMetadata, registeredUids.filter((token) => token !== htrUid));
```
Comment (pedroferreira1) [inline src/sagas/wallet.js:323]: What's this change?

**Review by pedroferreira1 — APPROVED** [review]

**Pull request merged**